platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java

```diff
@@ -341,16 +386,59 @@ public void onRestart() {
         super.onRestart();
     }
 
+    private void onItemMenuClick(View view, int position) {
+        Log.d(TAG, "onItemMenuClick ");
+
+        final QFieldProjectListItem item =
+            (QFieldProjectListItem)list.getAdapter().getItem(position);
+        File file = item.getFile();
+        PopupMenu popupMenu = new PopupMenu(QFieldProjectActivity.this, view);
+        popupMenu.setOnMenuItemClickListener(QFieldProjectActivity.this);
+
+        if (!file.isDirectory()) {
+            popupMenu.inflate(R.menu.project_item_menu);
+        } else {
+            String favoriteDirs =
+                sharedPreferences.getString("FavoriteDirs", null);
+            ArrayList<String> favoriteDirsArray = new ArrayList<String>();
+            if (favoriteDirs != null) {
+                favoriteDirsArray = new ArrayList<String>(
+                    Arrays.asList(favoriteDirs.split("--;--")));
+            }
+            boolean isFavorite = false;
+            for (String favoriteDir : favoriteDirsArray) {
+                if (favoriteDir.equals(file.getPath())) {
+                    isFavorite = true;
+                    break;
+                }
+            }
+
+            popupMenu.inflate(R.menu.project_folder_menu);
+            if (!isFavorite) {
+                popupMenu.getMenu()
+                    .findItem(R.id.remove_from_favorite)
+                    .setVisible(false);
+            } else {
+                popupMenu.getMenu()
+                    .findItem(R.id.add_to_favorite)
+                    .setVisible(false);
+            }
+        }
+
+        currentPosition = position;
+        popupMenu.show();
+    }
+
     private void onItemClick(int position) {
-        Log.d(TAG, "onListItemClick ");
+        Log.d(TAG, "onItemClick ");
```

Comment:
I'm sometimes wondering how much of this "I am here" logging we need. No strong opinion.